src/com/goide/intentions/GoMoveToStructAssignmentIntention.java

```diff
@@ -0,0 +1,238 @@
+    GoReferenceExpression selectedField = getFieldReference(element, assignment);
+    if (selectedField == null) return;
+    GoCompositeLit structDefinition = getStructDefinitionByField(selectedField, assignment);
+    String qualifierName = getQualifierName(selectedField);
+    if (structDefinition == null || qualifierName == null) return;
+    List<GoExpression> fields = getUninitializedFieldsByQualifier(assignment, structDefinition, qualifierName);
+    if (fields.isEmpty()) return;
+    moveFields(fields, assignment, structDefinition, project);
+  }
+
+  @NotNull
+  private static List<GoExpression> getUninitializedFieldsByQualifier(@NotNull GoAssignmentStatement assignment,
+                                                                      @NotNull GoCompositeLit definition,
+                                                                      @NotNull String qualifier) {
+    return ContainerUtil.filter(assignment.getLeftHandExprList().getExpressionList(),
+                                e -> isFieldDefinitionReference(e) &&
+                                     qualifier.equals(getQualifierName((GoReferenceExpression)e)) &&
+                                     isFieldUninitialized((GoReferenceExpression)e, definition));
+  }
+
+  private static void moveFields(@NotNull List<GoExpression> fields,
+                                 @NotNull GoAssignmentStatement fromAssignment,
+                                 @NotNull GoCompositeLit toStructDefinition,
+                                 @NotNull Project project) {
+    GoLiteralValue structFieldDefinitions = toStructDefinition.getLiteralValue();
+    if (structFieldDefinitions == null) return;
+    for (GoExpression expr : fields) {
+      if (!(expr instanceof GoReferenceExpression)) continue;
+      GoReferenceExpression field = (GoReferenceExpression)expr;
+      if (!isFieldDefinitionReference(field)) continue;
+
+      GoExpression fieldValue = GoPsiImplUtil.getExpressionValue(fromAssignment, field);
+      if (fieldValue == null) continue;
+
+      GoPsiImplUtil.deleteExpressionFromAssignment(fromAssignment, field);
+      addFieldDefinition(structFieldDefinitions, getFieldName(field), fieldValue.getText(), project);
+    }
+  }
+
+  @Nullable
+  private static String getQualifierName(@NotNull GoReferenceExpression fieldReference) {
+    GoReferenceExpression qualifier = fieldReference.getQualifier();
+    return qualifier != null ? qualifier.getText() : null;
+  }
+
+  private static boolean isAssignmentEqualsSized(@NotNull GoAssignmentStatement assignment) {
+    List<GoExpression> expressions = assignment.getExpressionList();
+    return assignment.getLeftHandExprList().getExpressionList().size() == expressions.size() &&
+           expressions.stream()
+             .allMatch(e -> !(e instanceof GoCallExpr) || GoInspectionUtil.getFunctionResultCount((GoCallExpr)e) == 1);
+  }
```

Comment:
Actually, i'm not quite sure how to handle these cases:

Duplicated fields: i think the best we can do is simply ignore this fields from moving/highlighting because this code is tricky and probably user don't knows what he want himself.

Second one: we can see, if `i` in struct init statement met only once and replace field value with corresponding `i` value with deleting `i` assignment, if it so. But this work is not too much related to `Move to struct assignment` logic, so i propose to not highlight this case too.

What do you think, can we do better in these situations?